src/Tribe/Tabbed_View.php

```diff
@@ -0,0 +1,366 @@
+<?php
+
+
+class Tribe__Tabbed_View {
+
+	/**
+	 * A list of all the tabs registered for the tabbed view.
+	 *
+	 * @var array An associative array in the [<slug> => <instance>] format.
+	 */
+	protected $items = array();
+
+	/**
+	 * The slug of the default tab
+	 *
+	 * @var string
+	 */
+	protected $default_tab;
+
+	/**
+	 * @var string The absolute path to this tabbed view template file.
+	 */
+	protected $template;
+
+	/**
+	 * An array or value object of data that should be used to render the tabbed view.
+	 *
+	 * @var array|object
+	 */
+	protected $data = array();
+
+	/**
+	 * @var string
+	 */
+	protected $label;
+
+	/**
+	 * @var string
+	 */
+	protected $url;
+
+	/**
+	 * @var string
+	 */
+	protected $active;
+
+	/**
+	 * Returns the tabbed view URL.
+	 *
+	 * @param array|string $args     Query String or Array with the arguments
+	 * @param boolean      $relative Return a relative URL or absolute
+	 *
+	 * @return string
+	 */
+	public function get_url( $args, $relative ) {
+		$relative_path = add_query_arg( $args, $this->url );
+
+		return $relative ? $relative_path : admin_url( $relative_path );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_template() {
+		return $this->template;
+	}
+
+	/**
+	 * @param string $template
+	 */
+	public function set_template( $template ) {
+		$this->template = $template;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_label() {
+		return $this->label;
+	}
+
+	/**
+	 * @param string $label
+	 */
+	public function set_label( $label ) {
+		$this->label = $label;
+	}
+
+	/**
+	 * Returns only the visible tabs for this tabbed view.
+	 *
+	 * @return Tribe__Tabbed_View__Tab[] An array of all the active and visible tabs.
+	 */
+	public function get_visibles() {
+		return array_filter( $this->get(), array( $this, 'is_tab_visible' ) );
+	}
+
+	/**
+	 * @param string $url
+	 */
+	public function set_url( $url ) {
+		$this->url = $url;
+	}
+
+	/**
+	 * Sets the slug of the currently active tab.
+	 *
+	 * This value, if the tab exists, will override the value specified in the GET request.
+	 *
+	 * @param string $active
+	 */
+	public function set_active( $active ) {
+		$this->active = $active;
+	}
+
+	/**
+	 * A method to sort tabs by priority in ascending order.
+	 *
+	 * @access private
```

Comment:
Do we need this (since the method signature tells us as much) and isn't it protected in any case?